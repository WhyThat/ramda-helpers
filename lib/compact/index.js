import { filter } from 'ramda';

/**
 * returns a list without falsy values
 * @name compact
 * @memberof RamdaHelper#
 * @param {Array}
 *
 * @returns {Array}
 * @example
 *
 * compact(['', 'test', 0, undefined, null])  //=> ['test']
 */
export default filter(el => !!el);
