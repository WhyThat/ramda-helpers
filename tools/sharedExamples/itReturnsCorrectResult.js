export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
