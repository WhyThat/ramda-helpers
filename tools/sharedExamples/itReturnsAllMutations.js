import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
