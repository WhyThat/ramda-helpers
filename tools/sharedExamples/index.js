
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
