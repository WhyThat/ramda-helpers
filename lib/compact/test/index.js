import compact from '../';
import { itReturnsAllMutations } from '../../test/sharedExamples';

describe('orderByDesc', () => {
  const mutations = [
    { name: 'With an empty list', props: [] },
    { name: 'With one falsey elements', props: [undefined] },
    { name: 'With elements', props: [1, '', 'test', null, 0, undefined] },
  ];

  itReturnsAllMutations(compact, mutations);
});
