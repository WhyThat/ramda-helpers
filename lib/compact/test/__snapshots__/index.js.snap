// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderByDesc With an empty list returns correct result 1`] = `Array []`;

exports[`orderByDesc With elements returns correct result 1`] = `
Array [
  1,
  "test",
]
`;

exports[`orderByDesc With one falsey elements returns correct result 1`] = `Array []`;
